src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10052,7 +10069,42 @@ private bool ScanParenthesizedImplicitlyTypedLambda(Precedence precedence)
                 return true;
             }
 
+            // Case 5: ( ! ) 
```

Comment:
>// Case 5: ( ! )  [](start = 12, length = 17)

Is it important to recognize this case? Is the error reporting or error recovery substantially better with this case? If not, consider removing it.